### PR TITLE
Some additional solid checks

### LINKED ISSUE #712: Construction banner accepted as a "solid" for solid placeholders
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you using the latest Structurize Version?

- [X] I am running the latest **beta/release** version of Structurize for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies/Structurize [Wiki](https://wiki.minecolonies.ldtteam.com/) or I asked on [discord](https://discord.minecolonies.com/).

### Minecraft Version

1.20

### Structurize Version

structurize-1.20.1-1.0.742-RELEASE

### MineColonies Version (if related bug)

minecolonies-1.20.1-1.1.740-snapshot

### Related Mods and their Versions

_No response_

### Current Behavior

Ray said to make a report:

When there are contruction tape placed by a schematic being built, in a spot that is set to be a solid placeholder in another schematic. The construction tape are accepted as a valid "solid" block by the builder.

### Expected Behavior

The builders break any construction tape that are in a spot set as Solid Placeholder.

### Reproduction Steps

Place two schematics, that have layers of solid placeholders, side by side, and start building one of them.

### Logs

None

### Anything else?

Nothing

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #712 
Closes #

# Changes proposed in this pull request
- Check if not replaceable/no collision for solidness too
-

## Testing
- [x] Yes I tested this before submitting it.
- [ ] I also did a multiplayer test.

Review please

### CODE REVIEW

**Comment by MotionlessTrain** [comment]
I assume this PR will close the issue, rather than itself? 😛

**Comment by Raycoms** [comment]
> I assume this PR will close the issue, rather than itself? 😛

Oui, I thought the issue was on minecolonies repo =D

**Comment by MotionlessTrain** [comment]
I mean, the link in the “closes” section mentions this PR, rather than the issue

**Comment by Raycoms** [comment]
> I mean, the link in the “closes” section mentions this PR, rather than the issue

oh, one number off xD

File: src/main/resources/META-INF/accesstransformer.cfg [hunk 1]
```diff
@@ -25,3 +25,5 @@ public net.minecraft.client.KeyMapping clickCount
 # itemHandler
 public net.minecraft.world.entity.decoration.GlowItemFrame getFrameItemStack()Lnet/minecraft/world/item/ItemStack;
 public net.minecraft.world.entity.decoration.ItemFrame getFrameItemStack()Lnet/minecraft/world/item/ItemStack;
+
+public net.minecraft.world.level.block.state.BlockBehaviour hasCollision # hasCollision
```
Comment (Nightenom) [inline src/main/resources/META-INF/accesstransformer.cfg:29]: do we need this? i mean if it isnt avaiable through sth else

**Review by Nightenom — COMMENTED** [review]
canBeReplaced makes somewhat sense to me
not sure if hasCollision is good idea

File: src/main/resources/META-INF/accesstransformer.cfg (hunk 1 above)
Comment (Raycoms) [reply src/main/resources/META-INF/accesstransformer.cfg:29]: it's the most efficient way to access it nicely.

**Comment by Raycoms** [comment]
> canBeReplaced makes somewhat sense to me not sure if hasCollision is good idea

Blocks that have no collision aren't solid for our intents and purposes. (e.g. players/entities) can't stand on them.

**Review by Nightenom — CHANGES_REQUESTED** [review]
This aint for "players/entities can stand of them" and shouldnt be used as such, if we're using it for this then we should have more definitions of solid
This solid is for building, ie.: if i place _it_ at 0,0,0 will it under all conditions stay at 0,0,0 unchanged? (unchanged wrt. blockstate and blockentity serialization data)

**Comment by Raycoms** [comment]
> This aint for "players/entities can stand of them" and shouldnt be used as such, if we're using it for this then we should have more definitions of solid This solid is for building, ie.: if i place _it_ at 0,0,0 will it under all conditions stay at 0,0,0 unchanged? (unchanged wrt. blockstate and blockentity serialization data)

It's not a "floating block placeholder" it's a solid placeholder. A block that has no collision block is not solid.

**Comment by Nightenom** [comment]
This changes I've done was for purpose of building being less error-prone, imho solid placeholders should really be tightened to sth like tag (and not to this rather weak check for building iterator)

**Review by someaddons — APPROVED** [review]

**Pull request merged**